src/CSharpExtDownloader.ts

```diff
@@ -9,34 +9,27 @@ import * as util from './common';
 import { Logger } from './logger';
 import { PackageManager, Status, PackageError } from './packages';
 import { PlatformInformation } from './platform';
+import { SetStatus, GetNetworkDependencies, ReportInstallationError, SendInstallationTelemetry } from './downloader.helper';
 
 /*
  * Class used to download the runtime dependencies of the C# Extension
  */
-export class CSharpExtDownloader
-{
+export class CSharpExtDownloader {
     public constructor(
         private channel: vscode.OutputChannel,
         private logger: Logger,
         private reporter: TelemetryReporter /* optional */,
         private packageJSON: any) {
     }
 
-    public installRuntimeDependencies(): Promise<boolean> {
+    public async installRuntimeDependencies(): Promise<boolean> {
         this.logger.append('Installing C# dependencies...');
+        this.logger.appendLine();
         this.channel.show();
 
-        let statusItem = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Right);
-        let status: Status = {
-            setMessage: text => {
-                statusItem.text = text;
-                statusItem.show();
-            },
-            setDetail: text => {
-                statusItem.tooltip = text;
-                statusItem.show();
-            }
-        };
+        let statusObject = SetStatus();
```

Comment:
Wow, the naming of this object is really tough with the current factoring!

How about we rename `SetStatus` to `GetRightAlignedStatusBarItem`? It's still not perfect, but we'll need to do deeper refactoring around this object to make it better, and that will be better done in a dedicated PR.